sampling/kvcache_model.py

```diff
@@ -87,7 +88,7 @@ def _generate_with_kvcache(self, prefix : torch.Tensor,
 
     @torch.no_grad()
     def generate(self, input : torch.Tensor, gamma : int) -> torch.Tensor:
-        output = self._generate_with_kvcache(input, gamma)
+        output = self._generate_with_kvcache(input.to(self._device), gamma)
```

Comment:
cpu-gpu communication harms performance and is unacceptable during inference.
I suggest we do not explicitly move tensors from CPU to GPU inside generate. This will make unexpected data moving during inference.